The 3-say_my_name Module:
========================
    
    
This module defines a function to print a formatted messagewith first and last names.

# Make Sure Documentation Exists:

    >>> say_my_name = __import__('3-say_my_name').say_my_name

    >>> module_documentation = __import__("3-say_my_name").__doc__
    >>> print(len(module_documentation) > 0)
    True

    >>> function_documentation = __import__("3-say_my_name").say_my_name.__doc__
    >>> print(len(function_documentation) > 0)
    True


# Correct Usage:

'say_my_name(...)': Print a formatted message with the provided first and last names.

    >>> say_my_name("John", "Smith")
    My name is John Smith
    
    >>> say_my_name("Don Quixotte")
    My name is Don Quixotte 
    
    >>> say_my_name("Walter", "White")
    My name is Walter White
    
    >>> say_my_name("Bob")
    My name is Bob 
    

# On Errors:

The parameters 'first_name' and 'last_name' must be strings. Otherwise, a TypeError is raised.
    
- Missing Argument Error:

    >>> say_my_name()
    Traceback (most recent call last):
    TypeError: say_my_name() missing 1 required positional argument: 'first_name'
    
- Too Many Arguments Error:

    >>> say_my_name("Captain", "Jack", "Sparrow")
    Traceback (most recent call last):
    TypeError: say_my_name() takes from 1 to 2 positional arguments but 3 were given
    
Non-String First Name Error:

    >>> say_my_name(50, "cent")
    Traceback (most recent call last):
    TypeError: first_name must be a string
    
    >>> say_my_name({"first_name": "James", "last_name": "Bond"})
    Traceback (most recent call last):
    TypeError: first_name must be a string

    >>> say_my_name(None, 0)
    Traceback (most recent call last):
    TypeError: first_name must be a string
    
    >>> say_my_name(1, 2)
    Traceback (most recent call last):
    TypeError: first_name must be a string
    
    >>> say_my_name("Heisenberg", 0)
    Traceback (most recent call last):
    TypeError: last_name must be a string
    
Non-String Last Name Error:

    >>> say_my_name("Man", ["With", "No" "Name"])
    Traceback (most recent call last):
    TypeError: last_name must be a string

    >>> say_my_name("Stumbly", (2, 3))
    Traceback (most recent call last):
    TypeError: last_name must be a string
